include/SFML/Window/ContextSettings.hpp

```diff
@@ -76,6 +81,8 @@ struct ContextSettings
     unsigned int majorVersion;      ///< Major number of the context version to create
     unsigned int minorVersion;      ///< Minor number of the context version to create
     Uint32       attributeFlags;    ///< The attribute flags to create the context with
+    std::string  vendorName;        ///< Name of the vendor of the OpenGL implementation
+    std::string  rendererName;      ///< Name of the OpenGL renderer implementation
```

Comment:
I don't really know the codebase of SFML or your coding conventions, but here's an idea:

Make vendorName and rendererName private variables in the ContextSettings struct and GlContext a friend class of the ContextSettings struct. Then add getter methods for vendorName and rendererName in the ContextSettings struct, so application developers can explicitly get the values of vendorName and rendererName by calling these methods.